docs/setup/gke.tf

```diff
@@ -0,0 +1,136 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "null_resource" "previous" {}
+
+resource "time_sleep" "wait_120_seconds" {
+  depends_on = [null_resource.previous]
+
+  create_duration = "120s"
+}
+
+data "google_client_config" "default" {}
+
+provider "kubernetes" {
+  host                   = "https://${module.gke.endpoint}"
+  token                  = data.google_client_config.default.access_token
+  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
+}
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+module "gke" {
+  depends_on              = [time_sleep.wait_120_seconds, module.asm-vpc]
+  source                  = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
+  version                 = "~> 16.0"
+  project_id              = var.project_id
+  name                    = "asm-cluster"
+  release_channel         = var.gke_channel
+  region                  = var.region
+  zones                   = [var.zone]
+  network                 = var.vpc
+  subnetwork              = var.subnet_name
+  ip_range_pods           = "${var.subnet_name}-pod-cidr"
+  ip_range_services       = "${var.subnet_name}-svc1-cidr"
+  config_connector        = true
+  enable_private_endpoint = false
+  enable_private_nodes    = true
+  master_ipv4_cidr_block  = "172.16.0.0/28"
+  cluster_resource_labels = { "mesh_id" : "proj-${data.google_project.project.number}" }
+}
+module "workload_identity" {
+  source              = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
+  version             = "~> 16.0.1"
+  gcp_sa_name         = "cnrmsa"
+  cluster_name        = module.gke.name
+  name                = "cnrm-controller-manager"
+  location            = var.zone
+  use_existing_k8s_sa = true
+  annotate_k8s_sa     = false
+  namespace           = "cnrm-system"
+  project_id          = module.enable_google_apis.project_id
+  roles               = ["roles/owner"]
+}
+module "asm-vpc" {
+  source  = "terraform-google-modules/network/google"
+  version = "~> 3.0"
+
+  project_id   = var.project_id
+  network_name = var.vpc
+  routing_mode = "GLOBAL"
+
+  subnets = [
+    {
+      subnet_name   = var.subnet_name
+      subnet_ip     = var.subnet_ip
+      subnet_region = var.region
+    },
+  ]
+
+  secondary_ranges = {
+    "${var.subnet_name}" = [
+      {
+        range_name    = "${var.subnet_name}-pod-cidr"
+        ip_cidr_range = var.pod_cidr
+      },
+      {
+        range_name    = "${var.subnet_name}-svc1-cidr"
+        ip_cidr_range = var.svc1_cidr
+      },
+      {
+        range_name    = "${var.subnet_name}-svc2-cidr"
+        ip_cidr_range = var.svc2_cidr
+      },
+    ]
+  }
+
+  firewall_rules = [{
```

Comment:
Is it a single cluster/subnet setup here? If yes, do we need this `firewall_rules` section?

Reply:
Gotcha, but I'm thinking about a minimal setup that could be reproduced by customers as-is without any part not needed. Just to avoid confusion. With that said, I let you see if it's needed or if you still want to keep it like this nevertheless :)